lib/src/status.rs

```diff
@@ -305,19 +307,141 @@ pub(crate) async fn status(opts: super::cli::StatusOpts) -> Result<()> {
     let mut out = out.lock();
     let legacy_opt = if opts.json {
         OutputFormat::Json
+    } else if std::io::stdout().is_terminal() {
+        OutputFormat::HumanReadable
     } else {
         OutputFormat::Yaml
     };
     let format = opts.format.unwrap_or(legacy_opt);
     match format {
         OutputFormat::Json => serde_json::to_writer(&mut out, &host).map_err(anyhow::Error::new),
         OutputFormat::Yaml => serde_yaml::to_writer(&mut out, &host).map_err(anyhow::Error::new),
+        OutputFormat::HumanReadable => human_readable_output(&mut out, &host),
     }
     .context("Writing to stdout")?;
 
     Ok(())
 }
 
+fn human_readable_output(mut out: impl Write, host: &Host) -> Result<()> {
+    for (status_string, status) in [
+        ("staged", &host.status.staged),
+        ("booted", &host.status.booted),
+        ("rollback", &host.status.rollback),
+    ] {
+        if let Some(host_status) = status {
+            if let Some(image) = &host_status.image {
+                writeln!(
+                    out,
+                    "Current {} image: {}",
+                    status_string, image.image.image
+                )?;
+
+                let version = image
+                    .version
+                    .as_deref()
+                    .unwrap_or("No image version defined");
+                let timestamp = image
+                    .timestamp
+                    .as_ref()
+                    .map(|t| t.to_string())
+                    .unwrap_or_else(|| "No timestamp present".to_owned());
+                let transport = &image.image.transport;
+                let digest = &image.image_digest;
+
+                writeln!(out, "    Image version: {version} ({timestamp})")?;
+                writeln!(out, "    Image transport: {transport}")?;
+                writeln!(out, "    Image digest: {digest}")?;
+            } else {
+                writeln!(out, "No {status_string} image present, checksum defined")?;
+            }
+        } else {
+            writeln!(out, "No {status_string} image present")?;
+        }
+    }
+    Ok(())
+}
+
+#[test]
+fn test_human_readable_base_spec() {
+    // Tests Staged and Booted, null Rollback
+    let spec_fixture: &str = include_str!("fixtures/spec-staged-booted.yaml");
```

Comment:
One thing I think would help here is to factor out a helper function like

`fn human_status_from_spec_fixture(&str) -> Result<String>` or so

that gets used in each test.

Reply:
Implemented a helper function, Following my interpretation of what you suggested. I used `.expect` to handled the result, should I instead be using the `.unwrap_or` that I used to handle the option types?